snuba/datasets/storages/events.py

```diff
@@ -315,6 +320,32 @@ def get_promoted_tags() -> Mapping[str, Sequence[str]]:
         # in the storage selector.
         ReadOnlyTableSelector("sentry_dist", "sentry_dist_ro"),
         EventsColumnProcessor(),
+        MappingColumnPromoter(
+            columns=all_columns,
+            mapping_spec={
+                "tags": PromotedColumnsSpec(
+                    key_field="key",
+                    val_field="value",
+                    column_mapping=ChainMap(
+                        {col.flattened: col.flattened for col in promoted_tag_columns},
+                        {
+                            col.flattened.replace("_", "."): col.flattened
+                            for col in promoted_context_tag_columns
+                        },
+                    ),
```

Comment:
you definitively do not want to know what happens here. It took me ages to understand why this worked at all in the implementation, and I spent a long time to find out when it broke (it did not).

Essentially these are the rules for tags promotion in the events dataset:
1) promoted_tag_columns preserve the same name (tag - column). These are not even passed to the current tag promotion code, because it has a default: if a tag X is requested and a column X exists between the promoted columns, then it is promtoed. No explicit mapping (https://github.com/getsentry/snuba/blob/master/snuba/datasets/tags_column_processor.py#L148)
2) promoted_context_tag_columns work the other way around. They have an explicit mapping, but the mapping is not expressed explicitly. The rule is that a column in the tag is turned into an underscore when looking for the promoted column.

Why are promoted_context_tag columns promtoed as tags instead of contexts is a mistery to me